include/picongpu/plugins/binning/axis/LogAxis.hpp

```diff
@@ -0,0 +1,287 @@
+
+                    BinWidthKernel(pmacc::HostDeviceBuffer<T, 1>& widths)
+                        : widthsDeviceBox{widths.getDeviceBuffer().getDataBox()}
+                    {
+                    }
+
+                    ALPAKA_FN_ACC T getBinWidth(uint32_t idx) const
+                    {
+                        return widthsDeviceBox(idx);
+                    }
+                };
+
+                LogAxis(
+                    AxisSplitting<T_Attribute> axSplit,
+                    T_AttrFunctor attrFunctor,
+                    std::string label,
+                    std::array<double, 7> units) // add type T to the default label string
+                    : axisSplit{axSplit}
+                    , label{label}
+                    , units{units}
+                    , lAK{attrFunctor, axisSplit.enableOverflowBins}
+                {
+                    initLAK();
+
+                    binWidths.reserve(axisSplit.nBins);
+                    // Calculate the logarithmic spacing factor
+                    T factor = std::pow(2., (lAK.logMax - lAK.logMin) / axisSplit.nBins);
+                    // Calculate bin widths
+                    T edge = axisSplit.m_range.min;
+                    for(int i = 0; i < axisSplit.nBins; ++i)
+                    {
+                        T next_edge = factor * edge;
+                        binWidths.emplace_back(next_edge - edge);
+                        edge = next_edge;
+                    }
+                }
+
+                void initLAK()
+                {
+                    // do conversion to PIC units here, if not auto
+                    // auto picRange = toPICUnits(axSplit.range, units);
+                    auto min = toPICUnits(axisSplit.m_range.min, units);
+                    PMACC_VERIFY(0. < min);
+                    auto logMin = std::log2(min);
+                    auto logMax = std::log2(toPICUnits(axisSplit.m_range.max, units));
+
+                    // do scaling calc here, on host and save it
+                    auto scaling = static_cast<decltype(logMax)>(axisSplit.nBins) / (logMax - logMin);
+
+                    auto nBins = axisSplit.nBins;
+                    if(axisSplit.enableOverflowBins)
+                    {
+                        nBins += 2;
+                    }
+
+                    lAK.initAxisSplit(logMin, logMax, nBins, scaling);
+                }
+
+
+                constexpr uint32_t getNBins() const
+                {
+                    return lAK.nBins;
+                }
+
+                double getUnitConversion() const
+                {
+                    return get_conversion_factor(units);
+                }
+
+
+                LogAxisKernel getAxisKernel() const
+                {
+                    return lAK;
+                }
+
+                BinWidthKernel getBinWidthKernel()
+                {
+                    // reset whenever binWidths changez
```

Comment:
Currently only a placeholder. This will only happen once runtime axis resizing is implemented, then this set flag will be moved out and be invalidated when the axis is resized.